backend/baseclass.pm

```diff
@@ -589,6 +589,16 @@ sub capture_screenshot {
     return;
 }
 
+sub reload_needles {
+    # called from testapi::set_var, so read the vars
+    bmwqemu::load_vars();
+
+    for my $n (needle->all()) {
+        $n->unregister();
+    }
```

Comment:
you don't like something like `map { $_->unregister } needle->all()`? - if that even works. I know, you just moved that code

Reply:
No, I don't like calling map for something that doesn't map

Reply:
got it